Crossed_Miner/Miner/MinerViewModel.cs

```diff
@@ -72,9 +78,27 @@ private void StartMining()
         {
             if (!System.IO.File.Exists(exeName))
             {
-                MessageBox.Show(Application.Current.MainWindow, "You must add t-rex.exe to this folder before mining.", "You Suck", MessageBoxButton.OK, MessageBoxImage.Information);
-                IsMining = false;
-                return;
+                MessageBoxResult result =  MessageBox.Show(Application.Current.MainWindow, "T-Rex.exe not found. Do you want to download it?", "You Suck", MessageBoxButton.YesNo, MessageBoxImage.Question);
+                switch(result)
+                {
+                    case MessageBoxResult.Yes:
+                        // Download T-Rex
+                        using (WebClient wc = new WebClient())
+                        {
+                            wc.UseDefaultCredentials = true;
+                            wc.DownloadProgressChanged += wc_DownloadProgressChanged;
+                            wc.DownloadFileCompleted += new AsyncCompletedEventHandler(wc_DownloadFileCompleted); ;
+                            wc.DownloadFileAsync( new System.Uri("https://github.com/trexminer/T-Rex/releases/download/0.19.9/t-rex-0.19.9-win-cuda11.1.zip"), "./t-rex.zip");
+                        }
+
+                        break;
+
+                    case MessageBoxResult.No:
+                    default:
+                        MessageBox.Show(Application.Current.MainWindow, "No mining for you then!", "Really Suck", MessageBoxButton.OK, MessageBoxImage.Error);
+                        isMining = false;
```

Comment:
`IsMining` is never set to true. In `StopMining()`  it should also be set to false.